static/js/services/playsound.js

```diff
@@ -22,141 +22,148 @@
 
-	};
+		var SoundInterval = function(sound, id, time) {
+			this.sound = sound;
+			this.id = id;
+			this.interval = null;
+			this.time = time;
+		};
+		SoundInterval.prototype.start = function() {
+			if (this.interval !== null) {
+				return;
+			}
+			var id = this.id;
+			var player = _.bind(function() {
+				return this.sound.play(id);
+			}, this);
+			player();
+			this.interval = setInterval(player, this.time);
+		};
+		SoundInterval.prototype.stop = function() {
+			clearInterval(this.interval);
+			this.interval = null;
+			delete this.sound.intervals[this.id];
+		};
 
-	Sound.prototype.initialize = function(options, aliases) {
+		var Sound = function(options, aliases) {
 
-		// Kill all the existing stuff if any.
-		if (this.sound) {
-			this.sound.stop();
-		}
-		_.each(this.intervals, function(i) {
-			i.stop();
-		});
-		this.intervals = {};
-
-		// Add error handler.
-		var onloaderror = options.onloaderror;
-		options.onloaderror = function(event) {
-			console.error("Failed to load sounds", event);
-			if (onloaderror) {
-				onloaderror.apply(this, arguments);
+			this.sound = null;
+			this.intervals = {};
+			if (options) {
+				this.initialize(options, aliases);
 			}
+
 		};
 
-		// Replace urls with their require generated URLs.
-		var urls = options.urls;
-		if (urls) {
-			var new_urls = [];
-			_.each(urls, function(u) {
-				u = require.toUrl(u);
-				new_urls.push(u);
+		Sound.prototype.initialize = function(options, aliases) {
+
+			// Kill all the existing stuff if any.
+			if (this.sound) {
+				this.sound.stop();
+			}
+			_.each(this.intervals, function(i) {
+				i.stop();
 			});
-			options.urls = new_urls;
-		}
+			this.intervals = {};
+
+			// Add error handler.
+			var onloaderror = options.onloaderror;
+			options.onloaderror = function(event) {
+				console.error("Failed to load sounds", event);
+				if (onloaderror) {
+					onloaderror.apply(this, arguments);
+				}
+			};
+
+			// Replace urls with their require generated URLs.
+			var urls = options.urls;
+			if (urls) {
+				var new_urls = [];
+				_.each(urls, function(u) {
+					u = require.toUrl(u);
+					new_urls.push(u);
+				});
+				options.urls = new_urls;
+			}
 
-		// Create the new shit.
-		this.players = {};
-		this.aliases = _.extend({}, aliases);
-		this.sound = new Howler.Howl(options);
+			// Create the new shit.
+			this.players = {};
+			this.aliases = _.extend({}, aliases);
+			this.sound = new Howler.Howl(options);
 
-		return this;
+			return this;
 
-	};
+		};
 
-	Sound.prototype.getId = function(id) {
+		Sound.prototype.getId = function(id) {
 
-		if (this.aliases.hasOwnProperty(id)) {
-			return this.aliases[id];
-		}
-		return id;
+			if (this.aliases.hasOwnProperty(id)) {
+				return this.aliases[id];
+			}
+			return id;
 
-	};
+		};
 
+		Sound.prototype.play = function(id, interval, autostart) {
 
-	Sound.prototype.play = function(id, interval, autostart) {
+			if (!this.shouldPlaySound()) {
```

Comment:
Pass the id as parameter, so plugins can grab it when the function is overwritten.